extensions/http-curl/tests/ControllerServiceIntegrationTests.cpp

```diff
@@ -126,34 +125,34 @@ int main(int argc, char **argv) {
   // now let's disable one of the controller services.
   std::shared_ptr<core::controller::ControllerServiceNode> cs_id = controller->getControllerServiceNode("ID");
   assert(cs_id != nullptr);
-  {
-    std::lock_guard<std::mutex> lock(control_mutex);
-    controller->disableControllerService(cs_id);
-    disabled = true;
-    waitToVerifyProcessor();
-  }
   {
     std::lock_guard<std::mutex> lock(control_mutex);
     controller->enableControllerService(cs_id);
     disabled = false;
-    waitToVerifyProcessor();
   }
   std::shared_ptr<core::controller::ControllerServiceNode> mock_cont = controller->getControllerServiceNode("MockItLikeIts1995");
-  assert(cs_id->enabled());
-{
+  const bool test_success_01 = verifyEventHappenedInPollTime(std::chrono::seconds(4), [&cs_id] {
+    return cs_id->enabled();
+  });
+  {
     std::lock_guard<std::mutex> lock(control_mutex);
     controller->disableReferencingServices(mock_cont);
     disabled = true;
-    waitToVerifyProcessor();
   }
-    assert(cs_id->enabled() == false);
-{
+  const bool test_success_02 = verifyEventHappenedInPollTime(std::chrono::seconds(2), [&cs_id] {
+    return !cs_id->enabled();
+  });
+  {
     std::lock_guard<std::mutex> lock(control_mutex);
     controller->enableReferencingServices(mock_cont);
     disabled = false;
-    waitToVerifyProcessor();
   }
-  assert(cs_id->enabled() == true);
+  const bool test_success_03 = verifyEventHappenedInPollTime(std::chrono::seconds(2), [&cs_id] {
```

Comment:
This seems to be exactly the first lambda recreated, can you simplify this code a bit? (an additional bool argument for the lambda as expected result and we just run the same func 3 times)

Reply:
Will extract them out into a single lambda function.